### PR TITLE
update ethereum and arbitrum tvl calculation

### PR DESCRIPTION
**Pull request by santinoYoung**
##### Chain:
ethereum
arbitrum

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/desyn-farm exports TVL: 
        
 
 
    merlin                    2.48 M
    
    total                    2.48 M

**Comment by llamatester** [comment]
The adapter at projects/desyn-farm1 exports TVL: 
        
 
 
    btr                       242.82 M
    core                      89.23 M
    mode                      43.27 M
    zklink                    22.57 M
    
    total                    397.90 M

**Comment by llamatester** [comment]
The adapter at projects/desyn exports TVL: 
        
 
 
    ethereum                  3.045991961850069e+53 B
    arbitrum                  257
    
    total                    3.045991961850069e+53 B

**Comment by llamatester** [comment]
The adapter at projects/desyn exports TVL: 
        
 
 
    ethereum                  3.0461193331482296e+53 B
    arbitrum                  257
    
    total                    3.0461193331482296e+53 B

**Review by g1nt0ki — CHANGES_REQUESTED** [review]
can you check the implementation. it is returning high tvl

**Comment by santinoYoung** [comment]
@g1nt0ki Hi, team.I'm sorry I submitted this commit.  but the calculation of TVL is done through the contract ledger based on business needs

**Comment by llamatester** [comment]
The adapter at projects/desyn exports TVL: 
        
 
 
    ethereum                  4.30 M
    arbitrum                  258
    
    total                    4.30 M

**Comment by santinoYoung** [comment]
@g1nt0ki  Hi, Team, I checked the pool assets and it happened that aSTETH has some special, temporary

**Comment by g1nt0ki** [comment]
@santinoYoung are the tokens going to be moved out for more yield somewhere else, or will be remain in this contract?

**Comment by santinoYoung** [comment]
> @santinoYoung are the tokens going to be moved out for more yield somewhere else, or will be remain in this contract?

It will stay here, but the contract design is special and the ledger gives the maximum value, so aSTETH is currently excluded for the time being

**Comment by llamatester** [comment]
The adapter at projects/desyn-farm1 exports TVL: 
        
 
 
    btr                       242.48 M
    core                      110.66 M
    mode                      43.61 M
    zklink                    22.67 M
    
    total                    419.42 M

**Comment by llamatester** [comment]
The adapter at projects/desyn exports TVL: 
        
 
 
    ethereum                  4.29 M
    arbitrum                  236
    
    total                    4.29 M

**Comment by santinoYoung** [comment]
The last commit should read：bitlayer delete doublecounted

File: projects/desyn-farm1/index.js [hunk 1]
```diff
@@ -2,7 +2,6 @@ const { getLogs } = require('../helper/cache/getLogs')
 
 module.exports = {
   methodology: 'On-chain restaking',
-  doublecounted: true
```
Comment (g1nt0ki) [inline projects/desyn-farm1/index.js:5]: if the tokens are not kept in the contract but moved outside, it should be marked as doublecounted since it's tvl is already counted towards chain tvl as part of the other protocol

File: projects/desyn/index.js [hunk 2]
```diff
@@ -33,7 +43,19 @@ Object.keys(config).forEach(chain => {
 
       const pools = logs.map(i=>i.pool)
       const tokens = await api.multiCall({  abi: 'address[]:getCurrentTokens', calls: pools})
-      return api.sumTokens({ ownerTokens: tokens.map((tokens, i) => [tokens, pools[i]])})
+      const calls = []
```
Comment (g1nt0ki) [inline projects/desyn/index.js:46]: again, are the tokens moving out of the contract here? right now, they remain in the contract, so why this change?

Comment (santinoYoung) [reply projects/desyn/index.js:46]: Previously it was calculated via balanceof, now it is via the contract's getBalance

**Comment by santinoYoung** [comment]
> The adapter at projects/desyn exports TVL:
> 
> ```
> ethereum                  3.0461193331482296e+53 B
> arbitrum                  257
> 
> total                    3.0461193331482296e+53 B 
> ```

If this token is not excluded, there is an error tvl

File: projects/desyn-farm1/index.js (hunk 1 above)
Comment (santinoYoung) [reply projects/desyn-farm1/index.js:5]: Okay, I'll re-add it doublecounted

**Comment by llamatester** [comment]
The adapter at projects/desyn exports TVL: 
        
 
 
    ethereum                  4.29 M
    arbitrum                  236
    
    total                    4.29 M

File: projects/desyn/index.js (hunk 2 above)
Comment (santinoYoung) [reply projects/desyn/index.js:46]: So for the time being this 0x1982b2F5814301d4e9a8b0201555376e62F82428token is not used as a tvl reference, any more questions here?
